packages/core-ethereum/src/transaction-manager.spec.ts

```diff
@@ -94,4 +94,18 @@ describe('transaction-manager', function () {
     expect(Array.from(transactionManager.confirmed.keys())).to.not.include(txs[0][0])
     expect(Array.from(transactionManager.confirmed.keys())).to.not.include(txs[1][0])
   })
+
+  it('should getAllPendingTxs', function () {
```

Comment:
Test for provider error handling missing.

Reply:
It's quite hard to test the case when provider emits `SERVER_ERROR` etc as it's not easy to spy on private property. As communicated with @jjperezaguinaga, he will handle the testing in https://github.com/hoprnet/hoprnet/issues/3013